design/one-pager-namespace-restricted-mode.md

```diff
@@ -0,0 +1,90 @@
+# Namespace-restricted mode for Crossplane Core
+
+* Owner: François Rigaut (@frigaut-orange)
+* Reviewers: Crossplane Maintainers
+* Status: Draft
+
+## Background
+
+The core Crossplane pod currently requires rights on Services, ServiceAccounts, 
+Deployments, and Secrets. If the pod does not have these permissions, it will
+not start properly. Either it will crash, or it will run without being able to
+reconcile any resource. This is due to the core's manager's cache watching
+resources in all namespaces by default.
+
+Additionnaly, Crossplane will attempt to create events on cluster-scoped resources
+in the `default` namespace. If it does not have the rights to do so, logs can be
+polluted by error messages.
+
+The crossplane ServiceAccount having this level of permissions may represent a 
+security issue. It may cause trouble for teams who manage a Crossplane instance
+on a cluster, without having full admin rights on it. This is why this proposal 
+introduces a new option to enable compatibility between Crossplane and permissions
+restricted only to its own namespace.
+
+## Goals
+
+The objective is for Crossplane to work properly with permissions on:
+
+- Cluster-scoped resources **only** for Crossplane custom resources
+- Namespaced-scoped kubernetes-native resources **only** in Crossplane's own 
+  namespace
+- Namespaced-scoped Crossplane custom resources in all namespaces (crossplane v2)
+
+If Crossplane is able to run with these minimal permissions, it enables user-side
+(i.e. outside of Crossplane) fine-tuning of the ClusterRole, in order to better 
+control which resources Crossplane can manipulate or not, and in which namespaces.
+
+## Minimal set of cluster permissions
+
+Here is the minimal desired set of cluster-wide permissions for crossplane in
+namespace-restricted mode:
+
+```yaml
+# ClusterRole
+rules:
+- apiGroups:
+  - apiextensions.crossplane.io
+  - pkg.crossplane.io
+  - secrets.crossplane.io
+  resources:
+  - "*"
+  verbs:
+  - "*"
+# This could even be optional depending on the desired level of control
+- apiGroups:
+  - apiextensions.k8s.io
+  resources:
+  - customresourcedefinitions
+  - customresourcedefinitions/status
+  verbs:
+  - "*"
+```
+
+Assuming crossplane has admin rights on all resources in its own namespace (or at
+least rights that are currently defined in the ClusterRole, except in a Role instead)
```

Comment:
The RBAC manager also has rights to make ClusterRole resources:

```
- apiGroups:
  - rbac.authorization.k8s.io
  resources:
  - clusterroles
  verbs:
  - bind
- apiGroups:
  - rbac.authorization.k8s.io
  resources:
  - clusterrolebindings
  verbs:
  - "*"
- apiGroups:
  - ""
  - coordination.k8s.io
  resources:
  - configmaps
  - leases
  verbs:
  - get
  - list
  - create
  - update
  - patch
  - watch
  - delete
```

The proposal should look at addressing how the RBAC manager should change? One option could be strict mode where the tooling installs Aggregated RBAC roles into the cluster while installing providers/XRDs?

Reply:
Ah, I see: 

```
1. Disable the RBAC manager pod, since the objective is to manage permissions 
   specific to an ecosystem (i.e. outside of Crossplane).
```